src/object/srv_obj_migrate.c

```diff
@@ -1257,22 +1258,26 @@ migrate_fetch_update_bulk(struct migrate_one *mrone, daos_handle_t oh,
 			D_GOTO(out, rc);
 	}
 
-	/* The data, rebuilt from replication, there are two cases,
-	 * 1. If it comes from overwritten, then it should have higher epoch than the above
-	 * data, rebuilt from parity.
-	 * 2. If it just partial update, then its epoch should not higher than the highest replicate
-	 * epoch on the parity shard.
-	 *
-	 * So let's choose minimum epoch + 1 as its update epoch, which is higher than case 1 epoch,
-	 * and also in case 2, there should have multiple different replicate recxs with different
-	 * epoch, so min_epoch + 1 should not be higher than the highest epoch anyway.
+	/* The data, rebuilt from replication, needs to keep the same epoch during rebuild,
+	 * otherwise it may mess up the relationship between parity epoch vs data(paritied epoch)
```

Comment:
"mess up" just confirm that is it because the replica epoch possibly lower or higher than parity epoch? So need to rebuild the replica ext one by one with there own epoch?

Then above L1246's comment 
"Since the parity shard epoch should be higher or equal to the data shard epoch,
  so let's use the minimum epochs of all parity shards as the update epoch of
   this data shard."

Seems that is incorrect? why the parity shard epoch >= data shard epoch? 
thx

Reply:
>Seems that is incorrect? why the parity shard epoch >= data shard epoch? and is it correct for above case "mrone->mo_iods_num_from_parity > 0" use "mrone->mo_min_epoch"?

this is data shard rebuild,  so it use the minimum epoch from parity epoch, to make sure its epoch will not be higher than the parity epoch, otherwise it will cause degraded fetch failure, hence cause data corruption.

mrone->mo_iods_num_from_parity is for rebuilding data shard. or I miss your question?

Reply:
ok, thanks.